tarantool/dbapi.py

```diff
@@ -0,0 +1,227 @@
+        self.arraysize = 1
+        self._rows = None
+
+    def callproc(self, procname, *params):
+        """
+        Call a stored database procedure with the given name. The sequence of
+        parameters must contain one entry for each argument that the
+        procedure expects. The result of the call is returned as modified
+        copy of the input sequence. Input parameters are left untouched,
+        output and input/output parameters replaced with possibly new values.
+        """
+        raise NotSupportedError("callproc() method is not supported")
+
+    @property
+    def rows(self):
+        return self._rows
+
+    @property
+    def description(self):
+        # FIXME Implement this method please
+        raise NotImplementedError("callproc() method is not implemented")
+
+    def close(self):
+        """
+        Close the cursor now (rather than whenever __del__ is called).
+        The cursor will be unusable from this point forward; DatabaseError
+        exception will be raised if any operation is attempted with
+        the cursor.
+        """
+        self._c = None
+        self._rows = None
+
+    def _check_not_closed(self, error=None):
+        if self._c is None:
+            raise ProgrammingError(error or "Can not operate on closed cursor")
+
+    def execute(self, query, params=None):
+        """
+        Prepare and execute a database operation (query or command).
+        """
+        self._check_not_closed("Can not execute on closed cursor.")
+
+        response = self._c.execute(query, params)
+
+        self._rows = response.rows
+        self._rowcount = response.rowcount
+        self._lastrowid = response.lastrowid
+
+    def executemany(self, query, param_sets):
+        self._check_not_closed("Can not execute on closed cursor.")
+        rowcount = 0
+        for params in param_sets:
+            self.execute(query, params)
+            if self.rowcount == -1:
+                rowcount = -1
+            if rowcount != -1:
+                rowcount += self.rowcount
+        self._rowcount = rowcount
+
+    @property
+    def lastrowid(self):
+        """
+        This read-only attribute provides the rowid of the last modified row
+        (most databases return a rowid only when a single INSERT operation is
+        performed).
+        """
+        return self._lastrowid
+
+    @property
+    def rowcount(self):
+        """
+        This read-only attribute specifies the number of rows that the last
+        .execute*() produced (for DQL statements like SELECT) or affected (
+        for DML statements like UPDATE or INSERT).
+        """
+        return self._rowcount
+
+    def _check_result_set(self, error=None):
+        """
+        Closed method for raising an error when Cursor object does not have
+        any row to fetch. Useful for checking access after DQL requests.
+        """
+        if self._rows is None:
+            raise ProgrammingError(error or "Result set is empty.")
```

Comment:
It is not empty, it does not exist at all. See the [example](https://stackoverflow.com/q/46301977/1598057) from MySQL Connector:

> mysql.connector.errors.InterfaceError: No result set to fetch from.

BTW, it gives InterfaceError in the case.

Reply:
Changed to InterfaceError and modified the error message